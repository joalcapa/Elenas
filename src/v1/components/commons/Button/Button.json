{
  "description": "Este componente renderiza un boton basico que ejecuta una funcion y se puede deshabilitar",
  "displayName": "Button",
  "methods": [],
  "props": {
    "onPress": {
      "type": {
        "name": "func"
      },
      "required": false,
      "description": "Funcion que ejecuta el boton cuando se hace click sobre el.",
      "defaultValue": {
        "value": "() => {}",
        "computed": false
      }
    },
    "text": {
      "type": {
        "name": "string"
      },
      "required": false,
      "description": "Nombre del boton.",
      "defaultValue": {
        "value": "''",
        "computed": false
      }
    },
    "isDefaultButton": {
      "type": {
        "name": "bool"
      },
      "required": false,
      "description": "Por defecto el boton es de color azul.",
      "defaultValue": {
        "value": "true",
        "computed": false
      }
    },
    "isGreenButton": {
      "type": {
        "name": "bool"
      },
      "required": false,
      "description": "El boton cambia a color verde.",
      "defaultValue": {
        "value": "false",
        "computed": false
      }
    },
    "isDisabled": {
      "type": {
        "name": "bool"
      },
      "required": false,
      "description": "Se deshabilita el boton.",
      "defaultValue": {
        "value": "false",
        "computed": false
      }
    }
  }
}