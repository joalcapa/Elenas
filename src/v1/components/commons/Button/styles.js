import {StyleSheet} from 'react-native';

import {Colors, FontSize} from '../../../resources';

const styles = StyleSheet.create({
    button_container: {
        borderRadius: '8px',
        paddingTop: '8px',
        paddingBottom: '8px',
        width: '100%', 
        maxWidth: '150px',
        displayFlex: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
    },
    text: {
        fontSize: FontSize.normal,
        color: Colors.C_000000,
    },
    default_button: {
        backgroundColor: Colors.C_4287f5,
    },
    green_button: {
        backgroundColor: Colors.C_42f575,
    },
    disabled_button: {
        backgroundColor: Colors.C_dbdbdb,
    },
});

export default styles;