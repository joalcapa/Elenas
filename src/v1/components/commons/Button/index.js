import React from 'react';
import PropTypes from 'prop-types';
import {TouchableOpacity, Text} from 'react-native';

import styles from './styles';

const Button = (props) => (
        <TouchableOpacity 
            onPress={props.onPress} 
            style={[
                styles.button_container,
                props.isDefaultButton ? styles.default_button : {},
                props.isGreenButton ? styles.green_button : {},
            ]}
        >
            <Text style={styles.text}>{props.text}</Text>
        </TouchableOpacity>
    );

Button.propTypes = {
    onPress: PropTypes.func,
    text: PropTypes.string,
    isDefaultButton: PropTypes.bool,
    isGreenButton: PropTypes.bool,
};

Button.defaultProps = {
    onPress: () => {},
    text: '',
    isDefaultButton: true,
    isGreenButton: false,
};

export default Button;