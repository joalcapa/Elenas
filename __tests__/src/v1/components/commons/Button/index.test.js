import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {create} from 'react-test-renderer';
import {TouchableOpacity} from 'react-native';
import {shallow, configure} from 'enzyme';

import Button from '../../../../../../src/v1/components/commons/Button';

configure({adapter: new Adapter()});
describe('Button component', () => {
    test('render blue button component', () => {
        const tree = create(<Button />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render green button component', () => {
        const tree = create(
            <Button
                isGreenButton
                onPress={() => {}}
                text="button title"
            />,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('onPress functionality', () => {
        const onPressEvent = jest.fn();
        onPressEvent.mockReturnValue('Link on press invoked');
        const wrapper = shallow(
            <Button onPress={onPressEvent} text="button title" />,
        );

        wrapper.find(TouchableOpacity).first().props().onPress();
        expect(onPressEvent.mock.results[0].value).toBe('Link on press invoked');
    });

    test('render disabled button component', () => {
        const tree = create(
            <Button
                isDisabled
                onPress={() => {}}
                text="button title"
            />,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('disabled functionality', () => {
        let isPress = true;
        const onPressEvent = jest.fn();
        onPressEvent.mockReturnValue('Link on press invoked');
        const wrapper = shallow(
            <Button isDisabled onPress={onPressEvent} text="button title" />,
        );

        try {
            wrapper.find(TouchableOpacity).first().props().onPress();
        } catch (error) {
            isPress = false;
        }

        expect(isPress).toBe(false);
    });
});