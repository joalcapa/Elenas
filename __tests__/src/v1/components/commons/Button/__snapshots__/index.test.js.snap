// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component render blue button component 1`] = `
<div
  className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-l7f4p8 r-borderRadius-1xfd6ze r-cursor-1loqt21 r-displayFlex-14n0d7j r-justifyContent-1777fci r-maxWidth-127gp16 r-paddingBottom-xd6kpl r-paddingTop-tskmnb r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr r-width-13qz1uu"
  onClick={[Function]}
  onContextMenu={[Function]}
  onKeyDown={[Function]}
  style={
    Object {
      "WebkitTransitionDuration": "0s",
      "transitionDuration": "0s",
    }
  }
  tabIndex="0"
>
  <div
    className="css-text-901oao r-color-cqee49 r-fontSize-1b43r93"
    dir="auto"
    onClick={[Function]}
  >
    
  </div>
</div>
`;

exports[`Button component render disabled button component 1`] = `
<div
  className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-oyv1ms r-borderRadius-1xfd6ze r-cursor-1loqt21 r-displayFlex-14n0d7j r-justifyContent-1777fci r-maxWidth-127gp16 r-paddingBottom-xd6kpl r-paddingTop-tskmnb r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr r-width-13qz1uu"
  onClick={[Function]}
  onContextMenu={[Function]}
  onKeyDown={[Function]}
  style={
    Object {
      "WebkitTransitionDuration": "0s",
      "transitionDuration": "0s",
    }
  }
  tabIndex="0"
>
  <div
    className="css-text-901oao r-color-cqee49 r-fontSize-1b43r93"
    dir="auto"
    onClick={[Function]}
  >
    button title
  </div>
</div>
`;

exports[`Button component render green button component 1`] = `
<div
  className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-15okr6x r-borderRadius-1xfd6ze r-cursor-1loqt21 r-displayFlex-14n0d7j r-justifyContent-1777fci r-maxWidth-127gp16 r-paddingBottom-xd6kpl r-paddingTop-tskmnb r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr r-width-13qz1uu"
  onClick={[Function]}
  onContextMenu={[Function]}
  onKeyDown={[Function]}
  style={
    Object {
      "WebkitTransitionDuration": "0s",
      "transitionDuration": "0s",
    }
  }
  tabIndex="0"
>
  <div
    className="css-text-901oao r-color-cqee49 r-fontSize-1b43r93"
    dir="auto"
    onClick={[Function]}
  >
    button title
  </div>
</div>
`;
